src/SymOp.jl

```diff
@@ -74,10 +74,41 @@ function check_group(symops::Vector; kwargs...)
             error("check_group: symop $s with inverse $(inv(s)) is not in the group")
         end
         for s2 in symops
-            if !is_approx_in_symops(s*s2) || !is_approx_in_symops(s2*s)
-                error("check_group: product is not stable")
+            if !is_approx_in_symops(s*s2)
+                error("check_group: product is not stable: $(s*s2) is not in the group")
             end
         end
     end
     symops
 end
+
+is_approx_in(group, symop; kwargs...) = any(s -> isapprox(s, symop; kwargs...), group)
+
+function complete_symop_group(symops; maxiter=10, kwargs...)
+    completed_group = Vector(symops)
+
+    for it = 1:maxiter
+        if it == maxiter
+            error("Could not complete group in $maxiter iterations")
+        end
+        to_add = []
+        function check(s1)
```

Comment:
Don't like the closure on `to_add` and implicit state change here, I'd move the function definition out of the for loop.
Regarding the name: I'd say the main effect is adding the missing elements and not checking something.